domain-proxy/Dockerfile.all-in-one

```diff
@@ -0,0 +1,15 @@
+FROM golang:alpine AS builder
+USER 0
+WORKDIR /work
+COPY ./ .
+
+RUN go build -o domain-proxy-server github.com/redhat-appstudio/jvm-build-service/domain-proxy/cmd/server
+RUN go build -o domain-proxy-client github.com/redhat-appstudio/jvm-build-service/domain-proxy/cmd/client
+
+FROM quay.io/konflux-ci/buildah-task:latest@sha256:5cbd487022fb7ac476cbfdea25513b810f7e343ec48f89dc6a4e8c3c39fa37a2
+USER 0
+RUN microdnf install -y iproute
+WORKDIR /work/
+
+COPY --from=builder /work/domain-proxy-server /app/domain-proxy-server
+COPY --from=builder /work/domain-proxy-client /app/domain-proxy-client
```

Comment:
Rather than splitting my comments across multiple locations which will get rather confusing I've embedded it all here. 

I am unclear as to why Dockerfile.all-in-one and Dockerfile.local are required?

In terms of the package layout I also have a some thoughts -

- Can we move the common code (e.g. pkg/client...) under the pkg structure (e.g. into pkg/domain-proxy/client/...)?
- Can the domain-proxy/cmd/client and  domain-proxy/cmd/server be moved under cmd (next to cmd/controller)?
- Do we need multiple go.mod files i.e. as the above simplifies the structure down? If the golang version needs to be 1.23 (See my other comment) we can always merge that associated PR.
- I don't think the executables need to be output straight into the working directory ; instead use similar to e.g.
```
go build -o out/jvmbuildservice
go build -o out/domain-proxy-server
```
   This then negates the `.gitignore` change
- In the Makefile I don't think there is a need to change `dev: dev-image-controller` ; instead move the golang compiling under the `build` target and the quay image generation  for both controller and domain-proxy can be under `dev-image`?

Reply:
> I am unclear as to why Dockerfile.all-in-one and Dockerfile.local are required?

'all-in-one' is used to build the domain proxy container as part of Github CI for use with minikube and buildah-oci-ta.
'local' is used to build the domain proxy container as part of local testing with minikube and buildah-oci-ta.

> Can we move the common code (e.g. pkg/client...) under the pkg structure (e.g. into pkg/domain-proxy/client/...)?
Can the domain-proxy/cmd/client and domain-proxy/cmd/server be moved under cmd (next to cmd/controller)?
Do we need multiple go.mod files i.e. as the above simplifies the structure down? If the golang version needs to be 1.23 (See my other comment) we can always merge that associated PR.

I don't think it makes sense to merge them all. They're separate applications and will be built differently. I could understand putting them all in the same structure if they directly depended on Go packages in the existing pkg structure. It's my understanding that a large amount of that existing Go code (maybe all?) will disappear at some point anyway with JBS disappearing.

> I don't think the executables need to be output straight into the working directory ; instead use similar to e.g. out
In the Makefile I don't think there is a need to change dev: dev-image-controller ; instead move the golang compiling under the build target and the quay image generation for both controller and domain-proxy can be under dev-image?

Both of these make sense to me. Will look at doing so.

Reply:
Ultimately I found it way easier to go with all of your suggestions. :smile: